File: src/baseDatos/Deserializador.java
package baseDatos;

import java.io.File;
import java.io.FileInputStream;
import java.io.ObjectInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

import gestorAplicacion.clasesPrincipales.*;

public class Deserializador {
    public static <E> void deserializador(List<E> list, String className) {
    	
        try {
            String path = System.getProperty("user.dir") + "/baseDatos/temp/" + className + ".txt"; // Ruta del archivo que se va a cargar
            File archivo = new File(path);  // Ruta donde crear este archivo, si aun no existe
            archivo.createNewFile(); // Crear un nuevo archivo, si no existe
            FileInputStream fileIn = new FileInputStream(path); // de donde cargar el archivo

            // Si el archivo esta vacio se lanza un throw EOFException y se muestra como un
            // mensaje de vacio, pero si no se usa el objeto in para leer el archivo
            ObjectInputStream in = new ObjectInputStream(fileIn);

            // Lee el listado de elementos
            //@SuppressWarnings("unchecked")
			ArrayList<E> listado = (ArrayList<E>) in.readObject();

            // Recorro el ArrayList
            for (E el : listado) {
                list.add(el);
            }

            in.close();
            fileIn.close();

        } catch (FileNotFoundException e) {
            System.out.println("No se encuentra archivo");
        } catch (IOException e) {
            System.out.println("Error flujo de inicialización");
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        }
        
    }

    //Las clases que se deben deserializar
    public static void deserializarTodo() {
        Deserializador.deserializador(Bus.getBuses(), "buses");
        Deserializador.deserializador(Conductor.getConductores(), "conductores");
        Deserializador.deserializador(Empresa.getEmpresas(), "empresas");
        Deserializador.deserializador(Pasajero.getPasajeros(), "pasajeros");
        Deserializador.deserializador(Ruta.getRutas(), "rutas");
        Deserializador.deserializador(Tiquete.getTiquetes(), "tiquetes");
        Deserializador.deserializador(Viaje.getViajes(), "viajes");

    }
}

File: Python/baseDatos/deserializador.py
import pickle
from gestor_aplicacion.clasesPrincipales.Tiquete import Tiquete
from gestor_aplicacion.clasesPrincipales.Bus import Bus
from gestor_aplicacion.clasesPrincipales.Pasajero import Pasajero
from gestor_aplicacion.clasesPrincipales.Ruta import Ruta
from gestor_aplicacion.clasesPrincipales.Viaje import Viaje
from gestor_aplicacion.clasesPrincipales.Conductor import Conductor
import pathlib
import os

class Deserializador:
    @classmethod
    def deserializar(cls, tipo):
        if tipo == "Bus":
            pickfile = open("baseDatos/temp/buses.txt", "rb")
            Bus._buses = pickle.load(pickfile)
            pickfile.close

        elif tipo == "Conductor":
            pickfile = open( "baseDatos/temp/conductores.txt", "rb")
            Conductor._conductores = pickle.load(pickfile)
            pickfile.close

        elif tipo == "Pasajero":
            pickfile = open("baseDatos/temp/pasajeros.txt", "rb")
            Pasajero._pasajeros = pickle.load(pickfile)
            pickfile.close

        elif tipo == "Ruta":
            pickfile = open("baseDatos/temp/rutas.txt", "rb")
            Ruta._rutas = pickle.load(pickfile)
            pickfile.close

        elif tipo == "Tiquete":
            pickfile = open("baseDatos/temp/tiquetes.txt", "rb")
            Tiquete._tiquetes = pickle.load(pickfile)
            pickfile.close

        elif tipo == "Viaje":
            pickfile = open("baseDatos/temp/viajes.txt","rb")
            Viaje.setViajes(pickle.load(pickfile))
            pickfile.close

    def DeserializarTodo():
        Deserializador.deserializar("Bus")
        Deserializador.deserializar("Conductor")
        Deserializador.deserializar("Pasajero")
        Deserializador.deserializar("Ruta")
        Deserializador.deserializar("Tiquete")
        Deserializador.deserializar("Viaje")
